app/controllers/transactions_controller.rb

```diff
@@ -0,0 +1,50 @@
+class TransactionsController < UsersController
+    before_action :find_user
+
+    def new
+        @transaction = Transaction.new
+        @account = @user.accounts.find(params[:transaction][:sender_id])
+    end
+
+
+    def create
+        @transaction = Transaction.new(transaction_params)
+        @account = @user.accounts.find(@transaction.sender_id)
+        if User.find_by(username: @transaction.receiver_name)
+            @receiver = User.find_by(username: @transaction.receiver_name)
+            if @receiver.accounts.find_by(account_name: @transaction.receiver_act)
+                @receiver_act = @receiver.accounts.find_by(account_name: @transaction.receiver_act)
+                if @account.balance >= @transaction.amount
+                    if @transaction.save
+                        @account.update(:balance => (@account.balance - @transaction.amount))
+                        @receiver_act.update(:balance => (@receiver_act.balance + @transaction.amount))
+                        flash[:success] = "Transaction Completed"
+                        redirect_to user_url(@user)
+                    else
+                        flash[:error] = @transaction.errors.full_messages
+                        render 'new'
+                    end
+                else
+                    flash[:notice] = "Insufficient Funds"
+                    render 'new'
+                end
+            else
+                flash[:notice] = "User does not have this account"
+                render 'new'
+            end
+        else
+            flash[:notice] = "User does not exist"
+            render 'new'
+        end
```

Comment:
This has a lot of logic in it and I think we can reduce and make this nicer to read by extracting all the checks and flash notices into their own methods.

So we could make the whole action look like
```
  def create
        @transaction = Transaction.new(transaction_params)
        @account = @user.accounts.find(@transaction.sender_id)
        @receiver = find_receiver
        @receiver_act = find_receiver_act
        if account_has_sufficient_funds && @transaction.save
           @account.update(:balance => (@account.balance - @transaction.amount))
           @receiver_act.update(:balance => (@receiver_act.balance + @transaction.amount))
           flash[:success] = "Transaction Completed"
           redirect_to user_url(@user)
        else
           flash[:error] = @transaction.errors.full_messages
           render 'new'
        end
    end
```

Reply:
then for example the `find_receiver` can be a private method and look like
```
def find_receiver
  receiver = User.find_by(username: @transaction.receiver_name)
  if receiver.present?
    return receiver
  else
    flash[:notice] = "User does not exist"
    render 'new'
  end
end
```

Reply:
or a slightly shorter version
```
def find_receiver
  receiver = User.find_by(username: @transaction.receiver_name)
  return receiver unless receiver.present?
  flash[:notice] = "User does not exist"
  render 'new'
end
```